dvc/logger.py

```diff
@@ -58,97 +52,57 @@ class ColorFormatter(logging.Formatter):
-            ).format(
-                color=self.color_code.get(record.levelname, ""),
-                nc=colorama.Fore.RESET,
-                levelname=record.levelname,
-                description=self._description(record.msg, exception),
-                msg=record.msg,
-                stack_trace=stack_trace,
+            return info
+
+        if record.exc_info:
+            _, exception, _ = record.exc_info
+
+            info = "{message}{separator}{exception}".format(
+                message=record.msg or "",
+                separator=" - " if record.msg and exception.args else "",
+                exception=": ".join(self._causes(exception)),
             )
 
-        return "{color}{levelname}{nc}: {msg}".format(
-            color=self.color_code.get(record.levelname, ""),
-            nc=colorama.Fore.RESET,
+            if self._current_level() == logging.DEBUG:
+                trace = "".join(traceback.format_exception(*record.exc_info))
+
+                return (
+                    "{red}{levelname}{nc}: {info}\n"
+                    "{red}{line}{nc}\n"
+                    "{trace}"
+                    "{red}{line}{nc}".format(
+                        levelname=record.levelname,
+                        info=info,
+                        red=colorama.Fore.RED,
+                        line="-" * 60,
+                        trace=trace,
+                        nc=colorama.Fore.RESET,
+                    )
+                )
+
+        return "{color}{levelname}{nc}: {info}".format(
+            color=self.color_code[record.levelname],
             levelname=record.levelname,
-            msg=record.msg,
+            nc=colorama.Fore.RESET,
+            info=info,
         )
 
-    def _current_level(self):
-        return logging.getLogger("dvc").getEffectiveLevel()
-
-    def _is_visible(self, record):
-        return record.levelno >= self._current_level()
-
-    def _description(self, message, exception):
-        description = ""
-
-        if exception and message:
-            description = "{message} - {exception}"
-        elif exception:
-            description = "{exception}"
-        elif message:
-            description = "{message}"
-
-        return description.format(message=message, exception=exception)
-
-    def _walk_exc(self, exc_info):
-        import traceback
-
-        buffer = io.StringIO()
-
-        traceback.print_exception(*exc_info, file=buffer)
-
-        exc = exc_info[1]
-        tb = buffer.getvalue()
-
-        exc_list = [str(exc)]
-        tb_list = [tb]
-
-        # NOTE: parsing chained exceptions. See dvc/exceptions.py for more info
-        while hasattr(exc, "__cause__") and exc.__cause__:
-            exc_list.append(str(exc.__cause__))
-            if hasattr(exc, "cause_tb") and exc.cause_tb:
-                tb_list.insert(0, str(exc.cause_tb))
+    def _causes(self, exc):
+        while exc:
+            yield str(exc)
```

Comment:
Could you please add a test for it, while we are at it? E.g. chanining 2 exceptions (we do have a test for 1 cause already)

Reply:
All set, I added it to the current `test_nested_exceptions`